src/arrayMethodJoin.js

```diff
@@ -5,8 +5,45 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let strFromArr = '';
+    const separatorSign = checkSeparator(separator);
+
+    if (this.length === 0) {
+      return strFromArr;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      const item = checkItem(this[i]);
+
+      if (i !== this.length - 1) {
+        strFromArr += item + separatorSign;
+      } else {
+        strFromArr += item;
+      }
+    }
+
+    return strFromArr;
   };
 }
 
+function checkSeparator(separator) {
```

Comment:
```suggestion
function checkSeparator(separator) {
```
No need in this function if you will just specify the default value for your `separator` argument